### PR TITLE
aggregates: add fetch cli

### PR DESCRIPTION
**Pull request by sileht**
This change adds the aggregates fetch API to the client.

### CODE REVIEW

File: gnocchiclient/v1/aggregates_cli.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+class CliAggregatesFetch(lister.Lister):
+    """Get measurements of aggregated metrics"""
+
+    COLS = ('timestamp', 'granularity', 'value')
+
+    def get_parser(self, prog_name):
+        parser = super(CliAggregatesFetch, self).get_parser(prog_name)
+        parser.add_argument("operations", required=True,
+                            help="Operations")
```
Comment (chungg) [inline gnocchiclient/v1/aggregates_cli.py]: Operations to apply to time series.

File: gnocchiclient/v1/aggregates_cli.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        parser.add_argument("--start",
+                            help="beginning of the period")
+        parser.add_argument("--stop",
+                            help="end of the period")
+        parser.add_argument("--granularity",
+                            help="granularity to retrieve")
+        parser.add_argument("--needed-overlap", type=float,
+                            help=("percent of datapoints in each "
```
Comment (chungg) [inline gnocchiclient/v1/aggregates_cli.py]: percentage of overlap across datapoints

File: gnocchiclient/v1/aggregates_cli.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                                  "metrics required"))
+        parser.add_argument("--groupby",
```
Comment (chungg) [inline gnocchiclient/v1/aggregates_cli.py]: is there groupby functionality?

File: gnocchiclient/v1/aggregates_cli.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+                            action='append',
+                            help="Attribute to use to group resources"),
+        parser.add_argument("--resample",
```
Comment (chungg) [inline gnocchiclient/v1/aggregates_cli.py]: this should be specified through operations?

File: setup.cfg [hunk 5]
```diff
@@ -70,6 +70,7 @@ openstack.metric.v1 =
     metric_measures_batch-metrics = gnocchiclient.v1.metric_cli:CliMetricsMeasuresBatch
     metric_measures_batch-resources-metrics = gnocchiclient.v1.metric_cli:CliResourcesMetricsMeasuresBatch
     metric_measures aggregation = gnocchiclient.v1.metric_cli:CliMeasuresAggregation
+    metric_aggregates_fetch = gnocchiclient.v1.metric_cli:CliAggregatesFetch
```
Comment (chungg) [inline setup.cfg]: metric_aggregates_fetch does seem slightly verbose now that i see it.

i think metric transform|translate|mutate are better. fetch seems weird to me now tbh.

**Comment by sileht** [comment]
Pull-request updated, HEAD is now b216d49c9a02d085bd787fc1b67b40dae2004da0

File: gnocchiclient/v1/aggregates_cli.py (hunk 3 above)
Comment (sileht) [reply gnocchiclient/v1/aggregates_cli.py]: Yes :)

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 469646fbc2de7cb05b40803b6747907c0cf23c81

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 81a3aeead6290cdf41eb7c90ce975e94ebd3d14d

File: gnocchiclient/tests/functional/test_aggregates.py [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+
+class AggregatesClientTest(base.ClientTestBase):
+    def test_scenario(self):
+        # PREPARE AN ARCHIVE POLICY
+        self.gnocchi("archive-policy", params="create agg-fetch-test "
+                     "--back-window 0 -d granularity:1s,points:86400")
+
+        self.gnocchi("resource", params="create metric-res1")
+        self.gnocchi("resource", params="create metric-res2")
+        print(self.gnocchi("resource", params="show metric-res2"))
```
Comment (chungg) [inline gnocchiclient/tests/functional/test_aggregates.py]: was this for testing?

**Comment by sileht** [comment]
Pull-request updated, HEAD is now efaa23a3435b53210f7ef5d8034f56be0c90482c

File: gnocchiclient/tests/functional/test_aggregates.py (hunk 6 above)
Comment (sileht) [reply gnocchiclient/tests/functional/test_aggregates.py]: damn, removed.

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 4e6c7c625ba7d7ac3afcfe7959ea9963a7cf8f67

**Pull request merged**